projects/observability/src/pages/explorer/explorer.component.ts

```diff
@@ -168,6 +191,23 @@ export class ExplorerComponent {
       this.initialState$.pipe(map(value => value.contextToggle.value.dashboardContext)),
       this.contextChangeSubject
     );
+    this.featureStateResolver.getFeatureState(ApplicationFeature.SavedQueries).subscribe(featureState => {
+      this.enableSavedQueries = featureState === FeatureState.Enabled ? true : false;
+    });
+  }
+
+  public onClickSaveQuery(): void {
+    this.preferenceService.get('savedQueries', []).subscribe(queries => {
+      this.savedQueries = queries as SavedQuery[];
+    });
+
+    const currentScope =
+      this.filters[0]?.metadata.scope === SPAN_SCOPE ? ScopeQueryParam.Spans : ScopeQueryParam.EndpointTraces;
+    const currentFilters = this.filters.map(filter => filter.urlString);
```

Comment:
Rather can getting the current scope this way, can we read the scope of the explorer by making `ExplorerGeneratedDashboardContext` a `BehaviourSubject` and getting its current value?

That will make this relatively easier to read and understand IMO. Happy to discuss around it.

Reply:
There's a `currentContext$` observable that already has the value of the scope. But we still need to do the ternary condition anyway because the context has different values than what we need exactly for the URL.

Values we get from currentContext$: `"API_TRACE"` or `"SPAN"`
Values we want: `"endpoint-traces"` or `"spans"`

Should I add a subscription to `currentContext$` in the constructor and put this ternary condition there?

Reply:
Just noticed there was already a method for this: `getQueryParamFromContext()`
Replaced the logic with that method here ✅